### ISSUE TITLE
X is unbound when X is imported in a module we're importing

### ISSUE THREAD

**Issue by SHxKM** [microsoft/pyright#293]
**Describe the bug**
Given file `base.py`:

```python
import os
```

And file `prod.py`

```python
from base import *
```

`os` is reported as unbound in `prod.py`.

**To Reproduce**
See above.

**Expected behavior**
I'm not sure what to suggest here. Maybe this *is* the expected behaviour? I mean surely `os` is available (everything is running correctly), but perhaps `pyright` is following the convention of not allowing these kind of bare imports (this is a legacy codebase and not much I can do to change now).

**Screenshots or Code**
See above.

**Comment by erictraut** [comment]
Let me make sure I understand your configuration. Are base.py and prod.py both source files in your program (i.e. under the root directory that you're asking pyright to analyze)? Or is base.py an external source file (one that's located outside of your root directory)?

**Comment by SHxKM** [comment]
base and prod are both source files. Both are nested under the project’s root directory.

**Comment by erictraut** [comment]
Huh, I'm not able to repro the problem. There must be something about your configuration that I'm not replicating. Could you provide more details? What is the directory layout? Are you using a pyrightconfig.json file? If so, what settings are you using? Do you have a python interpreter configured? Are you using the command-line version or the VS Code extension of pyright?

**Comment by SHxKM** [comment]
>  What is the directory layout?

`my_proj/settings/`:

```
base.py
dev.py
```

> Are you using a pyrightconfig.json file? If so, what settings are you using?

No.

> Do you have a python interpreter configured?

Yes

> Are you using the command-line version or the VS Code extension of pyright?

VSCode extension.

Note, I have downgraded my pyright version yesterday due to a separte (now fixed) bug (#296), and the above does not ocurr anymore. I will have time to report tonight whether the newest version also exmplifies this behaviour.

**Comment by erictraut** [comment]
OK, I suspect the problem you reported was the same as the one that I fixed (#296). If that's not the case, let me know and I'll dig into it.

**Comment by erictraut** [comment]
I'm going to assume this has been fixed. Feel free to open if you discover otherwise.
